### PR TITLE
endpoint configuration for s3 storage

### LINKED ISSUE #1773: Support for S3 api compatible storages
## Feature Request
It would be nice to add endpoint configuration option for S3 filesystem, to override default endpoint url. 

Modify `src/DependencyInjection/Configuration.php`
```
->arrayNode('s3')
    ->children()
        ...
        ->scalarNode('endpoint')->defaultNull()->end()
        ...
```

Modify `src/DependencyInjection/SonataMediaExtension.php`
```
public function configureFilesystemAdapter(ContainerBuilder $container, array $config)
    {
        ...
        if (isset($config['filesystem']['s3']['endpoint'])) {
            $arguments['endpoint'] = $config['filesystem']['s3']['endpoint'];
        }
        ...
```

### PR DESCRIPTION
**Pull request by michalpicpauer**
## Subject

Add endpoint configuration option for s3 filesystem.

I am targeting this branch, because these changes respect BC.

Closes #1773 https://github.com/sonata-project/SonataMediaBundle/pull/1706

## Changelog

```markdown
### Added
- Added configuration option filesystem.s3.endpoint
```

### CODE REVIEW

**Comment by michalpicpauer** [comment]
Ok, I made some changes. Is it fine to update the existing test with this new key?

**Comment by jordisala1991** [comment]
Yes please, update the tests and add a new one to test what happens when this new key is configured

**Comment by jordisala1991** [comment]
Plus it would be nice to update docs with that addition

**Comment by michalpicpauer** [comment]
> Yes please, update the tests and add a new one to test what happens when this new key is configured

The updated test should cover both situations or am I missing something?

File: tests/DependencyInjection/SonataMediaExtensionTest.php [hunk 1]
```diff
@@ -224,6 +226,7 @@ public function dataFilesystemConfiguration(): array
                             'sdk_version' => 3,
                             'region' => 'region',
                             'version' => 'version',
+                            'endpoint' => null,
```
Comment (jordisala1991) [inline tests/DependencyInjection/SonataMediaExtensionTest.php:229]: I think this line should be removed, people not using endpoint will not define it to null, they will simply not have it defined at all

File: tests/DependencyInjection/SonataMediaExtensionTest.php [hunk 2]
```diff
@@ -178,12 +178,14 @@ public function testLoadWithSonataAdminCustomConfiguration(): void
         $fakeContainer->expects($this->once())
             ->method('getExtensionConfig')
             ->with($this->equalTo('sonata_admin'))
-            ->willReturn([[
-                'security' => [
-                    'role_admin' => 'ROLE_FOO',
-                    'role_super_admin' => 'ROLE_BAR',
+            ->willReturn([
```
Comment (core23) [inline tests/DependencyInjection/SonataMediaExtensionTest.php:181]: Please revert. This is unrelated to this PR.

File: src/DependencyInjection/SonataMediaExtension.php [hunk 3]
```diff
@@ -399,6 +399,10 @@ public function configureFilesystemAdapter(ContainerBuilder $container, array $c
                     'version' => $config['filesystem']['s3']['version'],
                 ];
 
+                if (isset($config['filesystem']['s3']['endpoint'])) {
```
Comment (core23) [inline src/DependencyInjection/SonataMediaExtension.php:402]: Please add an other test that does not modify the endpoint and skips this line

Comment (jordisala1991) [reply src/DependencyInjection/SonataMediaExtension.php:402]: Isn't this one already skipping this line? https://github.com/sonata-project/SonataMediaBundle/pull/1774/files#diff-217ffd8f1233ee89cb0db792bd11981dL220-L236

Comment (jordisala1991) [reply src/DependencyInjection/SonataMediaExtension.php:402]: Only half of tests go inside the if

Captura de pantalla 2020-07-31 a las 14 01 30

**Comment by VincentLanglet** [comment]
Please review @core23 :)
Does the @jordisala1991 answer is enough for you ?

**Comment by jordisala1991** [comment]
Thank you @michalpicpauer
